libtracepoint-control-cpp/src/TracepointSession.cpp

```diff
@@ -1447,28 +1638,16 @@ TracepointSession::EnumeratorMoveNext(
 
         buffer.DataPos += eventHeader.size;
 
-        uint16_t const dataSize = eventHeader.size - sizeof(perf_event_header);
-        uint32_t const dataPos = (eventHeaderBufferPos + sizeof(perf_event_header)) & (m_bufferSize - 1);
-        if (eventHeader.type == PERF_RECORD_SAMPLE)
+        if (eventHeader.type == PERF_RECORD_LOST)
         {
-            if (sampleFn(bufferData, dataSize, dataPos))
-            {
-                return true;
-            }
+            auto const newEventsLost64 = *reinterpret_cast<uint64_t const*>(
+                bufferData + ((eventHeaderBufferPos + sizeof(perf_event_header) + sizeof(uint64_t)) & (m_bufferSize - 1)));
+            m_lostEventCount += newEventsLost64;
         }
-        else
-        {
-            if (eventHeader.type == PERF_RECORD_LOST)
-            {
-                auto const newEventsLost64 = *reinterpret_cast<uint64_t const*>(
-                    bufferData + ((eventHeaderBufferPos + sizeof(perf_event_header) + sizeof(uint64_t)) & (m_bufferSize - 1)));
-                m_lostEventCount += newEventsLost64;
-            }
 
-            if (nonSampleFn(bufferData, dataSize, dataPos, eventHeader))
-            {
-                return true;
-            }
+        if (recordFn(bufferData, eventHeader.size, eventHeaderBufferPos))
+        {
+            return true;
         }
```

Comment:
Note: Changed handling of the internal event enumeration system.
- No longer skipping the header before invoking the callback. Previously, the callback got the offset after the header and the size excluding the header. Now the callback gets the offset to the header and the size including the header.
- No longer using separate callbacks for SAMPLE and non-SAMPLE events. Just one callback for both types.